code/game/machinery/firealarm.dm

```diff
@@ -232,35 +232,35 @@ FIRE ALARM
 	return
 
 /obj/machinery/firealarm/proc/alarm(var/duration = 0)
-	if(!( src.working))
+	if(!( working))
 		return
 	var/area/A = get_area(src)
 	for(var/obj/machinery/firealarm/FA in A)
 		fire_alarm.triggerAlarm(loc, FA, duration)
 	update_icon()
-	//playsound(src.loc, 'sound/ambience/signal.ogg', 75, 0)
+	//playsound(loc, 'sound/ambience/signal.ogg', 75, 0)
 	return
 
 /obj/machinery/firealarm/New(loc, dir, building)
 	..()
 
 	if(loc)
-		src.loc = loc
```

Comment:
This is going to break things; this is a legitimate instance of implied `src` being necessary. 

Careful when removing `src` to check for things like this.